parser/fileparser.go

```diff
@@ -205,7 +206,21 @@ func mapBroHeaderToParserType(header *fpt.BroHeader, broDataFactory func() pt.Br
 //the BroData created by the broDataFactory
 func parseLine(lineString string, header *fpt.BroHeader,
 	fieldMap fpt.BroHeaderIndexMap, broDataFactory func() pt.BroData,
-	logger *log.Logger) pt.BroData {
+	isJSON bool, logger *log.Logger) pt.BroData {
+
+	dat := broDataFactory()
+
+	if isJSON {
+		err := json.Unmarshal([]byte(lineString), dat)
+		if err != nil {
+			logger.WithFields(log.Fields{
+				"error": err.Error(),
+			}).Error("Encountered unparsable JSON in log")
+		}
+		dat.ConvertFromJSON()
+		return dat
+	}
```

Comment:
If the line is coming from a json file, we short circuit the tsv parsing logic and use the `encoding/json` library to unmarshal the string to an object. Then to take care of the timestamp discrepancy we call the new `ConvertFromJSON()` function.

Reply:
I like the idea in theory. An issue I haven't worked out is how coupled fsimporter is with everything else in the parser package. Let me give it a shot and I'll come back if I think it will cause too many sweeping changes.